chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -257,10 +289,57 @@
         Chef::Log.info("#{name} usurped #{res[1].join(", ")} routes from #{bound_if}") unless res[1].empty?
       end
     end
+
+    execute "Neutron: add #{bound_if} to #{name}" do
+      command "ovs-vsctl add-port #{name} #{bound_if}"
+      not_if not_if_clause
+    end
     source = ::Nic.new(bound_if)
     # filter out cluster VIPs that a currently assigned to the interface
     addresses = source.addresses.select { |address| my_addresses.include? address }
     routes = source.routes
+
+    # Try to bring the cluster back up.
+    if handle_corosync_restart
+      service "start openais after network reconfiguration" do
+        service_name node[:corosync][:platform][:service_name]
+        action [:start]
+        not_if { addresses.empty? and routes.empty? }
```

Comment:
```not_if``` is evaluated during the convergence phase AFAIK. But I didn't want reuse the old ```not_if```, I just wanted to assign the new value to it (```addresses.empty? and routes.empty?```) cause it's also used in multiple places.

Regarding the corosync restart, AFAIK the corosync cookbook will think corosync wasn't shutdown cleanly and because of that refuses to start it again, but maybe I am getting this wrong. Additionally there is still the problem that the cluster will be in maintenance mode after that even if it got started …